enigma-runtime-t/src/lib.rs

```diff
@@ -185,6 +210,64 @@ impl Runtime {
         Ok(())
     }
 
+    fn treat_gas_overflow(&mut self, val: &Option<u64>) -> Result<()>{
+        if val.is_none() {
+            self.gas_counter = self.gas_limit;
+            Err(WasmError::GasLimit)
+        }
+        else {
+            Ok(())
```

Comment:
Maybe match over the `val` here and return the contained value in the `Ok()` then you won't need to later unwrap the val.